packages/langchain/src/openai/util.test.ts

```diff
@@ -0,0 +1,106 @@
+// mapResponseToChatResult
+// mapLangchainToAiClient
+
+import {
+  OpenAiChatClient as OpenAiChatClientBase,
+  OpenAiChatCompletionOutput
+} from '@sap-ai-sdk/foundation-models';
+import { jest } from '@jest/globals';
+import nock from 'nock';
+import { HumanMessage } from '@langchain/core/messages';
+import {
+  mockClientCredentialsGrantCall,
+  mockInference,
+  parseMockResponse
+} from '../../../../test-util/mock-http.js';
+import { mapResponseToChatResult } from './util.js';
+import { OpenAiChatClient } from './chat.js';
+
+const openAiMockResponse = parseMockResponse<OpenAiChatCompletionOutput>(
+  'foundation-models',
+  'openai-chat-completion-success-response.json'
+);
+
+const chatCompletionEndpoint = {
+  url: 'inference/deployments/1234/chat/completions',
+  apiVersion: '2024-02-01'
+};
+
+const prompt = {
```

Comment:
[pp] Can you specify the types for each variable, it's easier to understand which type is expected to transform into which.

Reply:
No, I imagined these are existing types right? For example, request would be type `OpenAiChatCompletionParameters` ?